src/main.rs

```diff
@@ -522,4 +529,9 @@ fn main() {
 		}
 	}).unwrap();
 	println!("\x1b[32;1mFinished.\x1b[m");
+
+	build_log.lock().unwrap().write(spec.build_dir.as_path().join(".ninja_log")).unwrap_or_else(|e| {
```

Comment:
No need to `lock()` the build_log, there is nobody else referencing it. See [`Mutex::get_mut`](https://doc.rust-lang.org/std/sync/struct.Mutex.html#method.get_mut).